### DISCUSSION TITLE
Connect a Squarespace domain to a website hosted on GitHub Pages

### DISCUSSION THREAD

**Discussion by nastja4** [community/community#78573] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello, I'm seeking assistance as I'm new to this and experiencing a delay in the DNS check on GitHub. I've provided the necessary data (screenshots), and it has been displaying 'DNS Check in Progress' for over 24 hours. Can someone help me identify if there's an issue with my data or actions? Thank you.
 
![Screenshot (485)](https://github.com/community/community/assets/126527606/1e66e18e-5a7b-482e-bbab-c22bc24417ae)
![Screenshot (488)](https://github.com/community/community/assets/126527606/90238acc-ee53-4771-9330-424d85ffe0a3)

**Comment by DylanDevelops** [comment]
I believe you are experiencing this because you didn't add all the IPs to your A records.

Please ensure a different A record for each of the following IPs:
`185.199.108.153`
`185.199.109.153`
`185.199.110.153`
`185.199.111.153`

After you create these, try adding your domain to the github pages and wait.

**Reply by nastja4** [reply]
Thanks Dylan. One developer told me, it's anough to specify only one of the IPs for A

**Reply by DylanDevelops** [reply]
Yeah, I don't believe this is true. Let me know if once you add these IPs, your problem is solved. If it is, make sure to mark this as the answer so others can find it if they have a similar issue!

**Reply by nastja4** [reply]
Ok, I added all 4x IPs for A records. I'm waiting then
![Screenshot (493)](https://github.com/community/community/assets/126527606/8516faeb-570c-4201-bc42-6a0221aeddd5)

**Reply by DylanDevelops** [reply]
Any updates? Does your domain seem to be working?

**Reply by nastja4** [reply]
Hi Dylan. No, I dont have any changes. I dont know, what should I do. I also asked Squarespace support about my DNS settings. I'm waiting for the feedback. I also have one strange issue I dont know how to interpret. When I reload the GitHub page I see an error. In 2 seconds as I click on "Pages" (on the left side) I see "DNS Check in Progress".

![Screenshot (500)](https://github.com/community/community/assets/126527606/efd3c120-5f4a-4270-a655-8da31cb4f36e)
![Screenshot (501)](https://github.com/community/community/assets/126527606/a6a8645c-9c2a-4b02-8cc0-c8155fddb565)

**Reply by DylanDevelops** [reply]
I'll send a screenshot of how I have www.dylanravel.com configured in the DNS:
<img width="674" alt="Screenshot 2023-12-11 at 11 30 04 AM" src="https://github.com/community/community/assets/48571264/430dd621-58fa-4a08-9e2e-72eedfd185e3">
Ignore the one `_domainconnect` DNS and look at how I have my subdomains configured. If your DNS still looks like that one previous screenshot, you may have the www configured incorrectly. Make sure you provide `www` for the name and not `www.example.com`.

it is also good to note that any future subdomains need different repositories. For example, your `www.example.com` is fine for this repository, but if in the future you want as an example `blog.example.com` you will need a new repository and configured for that CNAME.

**Reply by nastja4** [reply]
Thanks! I'l try that

**Reply by nastja4** [reply]
Now I see something new on GitHub page

![Screenshot (502)](https://github.com/community/community/assets/126527606/17eb823d-0c03-4b86-90c3-32dd79152efe)
![Screenshot (503)](https://github.com/community/community/assets/126527606/50c17f15-c892-45d8-aacc-ea21281292ef)
GitHub page

**Reply by DylanDevelops** [reply]
Looks like it is working. Give that some time. If it doesn't work after around 15 minutes, remove your domain and, add it again in your GitHub settings, wait another 15 minutes. try committing to the website so that it does another deploy. If, after this it still doesn't work, let me know!

**Reply by nastja4** [reply, marked as answer]
I have a new notification, so I've changed the host names to "@" for A points. But my web site https://www.anastasiafazius.com/ is working.

![Screenshot (506)](https://github.com/community/community/assets/126527606/69fe0469-de2f-4907-a6c7-026c35ccbfcf)
![Screenshot (505)](https://github.com/community/community/assets/126527606/a8767166-742d-4e70-b1cc-b30203fc486e)

**Reply by DylanDevelops** [reply]
Sweet your website looks like it is ready! DNS propagation is not immediate. 
According to [this](https://github.com/orgs/community/discussions/35168) discussion post, this is what they had to say about that error:

> GitHub will not be able to tell you if your DNS configuration is good until it resolves from our side of the Internet.

I think you are all set to go. Try committing to your website and see if it is updating. If it is, I would mark this discussion as the answer so that others can find it if they are having the same issue. Feel free to message again later if you have another issue with pages!

**Reply by nastja4** [reply]
I have committed the project multiple times, and the website data is successfully updated. Everything seems to be working smoothly. Thank you so much, Dylan!

**Reply by DylanDevelops** [reply]
Yes of course! Glad I could help you out. I remember when I was first getting my website set up it felt like it was a little undocumented. 

Just mark this thread as the answer!

**Comment by nastja4** [comment]
Thanks for your input and assistance. I've successfully connected the Squarespace domain to the GitHub Pages website using the following configurations:

1. **GitHub project Settings - Pages:**
   - Custom Domain: www.example.com

2. **Squarespace DNS Settings:**
   - Deleted default settings
   - CNAME Record: Set to 'www' pointing to 'username.github.io'
   - A Records: Four A records ('@') pointing to GitHub's IP addresses
  
```
185.199.108.153
185.199.109.153
185.199.110.153
185.199.111.153
```

Please note that the DNS changes may take some time to propagate. If you have any further questions or if there are any issues, feel free to open a new discussion.

Closing this thread for now.

**Reply by xefech** [reply]
My website redirects only with www at beginning. Is it okay?

**Reply by nastja4** [reply]
Hello Gaabriel. I can only say that I tried everything, but it worked for me only with these DNS settings:
![Screenshot (698)](https://github.com/community/community/assets/126527606/e5dc1260-04f9-4146-b37d-c58d91357721)

**Comment by IbrahimOezhan** [comment]
![image](https://github.com/community/community/assets/89545837/ccb35be8-e6ac-495a-815d-05a433d176fb)
Can someone help I am getting this error when trying to add this

**Reply by nastja4** [reply]
Hi. I would say, try to delete all the other records (what are these above "Custom records"?); add four A records to your CNAME record

**Reply by IbrahimOezhan** [reply]
That are the squarespace default records. And I added the 4 A records and then the CNAME one but it always throws this error when I try to add the CNAME

**Reply by nastja4** [reply]
Then delete default records. I did the same way back that time. See my screenshot of "DNS Settings" above?

**Reply by IbrahimOezhan** [reply]
Deleting the default fixed it! Thanks a lot. Seems it was preventing me from adding two CNAMES since there was a default one but without mentioning it in the error message

**Reply by nastja4** [reply]
Nice! You're welcome. Yep, the error message could have been more informative.

**Comment by waitzShigoto** [comment]
Are these IP addresses or servers permanently fixed?


185.199.108.153
185.199.109.153
185.199.110.153
185.199.111.153

Because I use the SQUARESPACE Default to add the recoard,
they show me different address.
![截圖 2024-10-16 晚上7 40 04](https://github.com/user-attachments/assets/5347b082-0a47-44b9-bf57-39121417005b)

**Reply by LiteBrite82** [reply]
yes

**Comment by tony9823** [comment]
Why is my DNS check on GitHub taking over 24 hours to complete? I've provided the required data, including screenshots, but it still displays 'DNS [Check](https://firehousubsmenu.us/) in Progress.' How can I determine if there's an issue with my setup or actions?

**Reply by nastja4** [reply]
Hi. If after 24 hours you still see the same status, something might be wrong with your data. Follow the instructions above, take a look at the screenshots to set up the connection correctly.

**Comment by serpico3** [comment]
![Schermata del 2025-03-10 10-59-04](https://github.com/user-attachments/assets/92213c67-836b-4650-afbb-606dabdeec3f)
![Schermata del 2025-03-10 10-58-48](https://github.com/user-attachments/assets/634e74a2-66d4-48fb-9497-096b7cfaa6f8)


I'm creating a website for a venue and I have the domain on Squarespace. I don't understand why it's not working; I believe I have configured everything correctly. The domain is: finotratto.it.

**Reply by nastja4** [reply]
Hi. There is no "www." in custom domain

**Reply by DylanDevelops** [reply]
> 

The domain you enter in the custom domain box must match this DNS record, including the host prefix.
<img width="583" alt="Screenshot 2025-03-10 at 7 53 57 AM" src="https://github.com/user-attachments/assets/60d6e217-ae6d-42f6-8dd2-307c0385300d" />

So it should be `www.finotratto.it` in the GitHub Pages custom domain box.

**Reply by nastja4** [reply]
![image](https://github.com/user-attachments/assets/5d0ff074-75b1-4a55-a5c3-350535415a51)
Must be "www.finotratto.it"

**Comment by Htavens** [comment]
Unsure what could be causing this error, does anyone have any advice?
![Screen Shot 2025-03-14 at 11 40 49 AM](https://github.com/user-attachments/assets/305e5423-7f04-4d8d-bfd5-1ae75e4422a9)
![Screen Shot 2025-03-14 at 11 40 58 AM](https://github.com/user-attachments/assets/d18c31e5-c82a-45ae-945b-09536188bf12)

**Reply by nastja4** [reply]
Hi. Is there really no a top level domain (.com/.net/.org part of a URL)?

**Reply by Htavens** [reply]
It's www.fandm-sakuramatsuri.net. It's typed in github, but I think it's just a bit long to all be displayed in the custom domains box.

**Reply by nastja4** [reply]
![Screenshot_20250314_171935_Chrome](https://github.com/user-attachments/assets/e9513707-b124-499a-ae66-a439838c223d)
I think you should try to set in DNS Settings htavens.github.io instead of Htavens....

**Comment by gmagno** [comment]
Hello!
Please I need a hand :pray: 

It's been like this for days.

Github:

<img width="1170" height="1097" alt="image" src="https://github.com/user-attachments/assets/b9def3c7-d124-4735-80ac-ca17659645dc" />

Squarespace:

<img width="1630" height="803" alt="image" src="https://github.com/user-attachments/assets/c9fcef1c-643e-4907-b4c6-2c719e87c724" />


The actual page when I try to open it:

<img width="1487" height="820" alt="image" src="https://github.com/user-attachments/assets/72d91bb9-78f6-45cf-94f9-b449eb427cdf" />

**Reply by nastja4** [reply]
Hi, maybe you forgot "www." in Custom domain section on Github page?

**Reply by nastja4** [reply]
<img width="301" height="131" alt="image" src="https://github.com/user-attachments/assets/a0e256be-d43c-4456-8db2-631781e74b8d" />

**Reply by gmagno** [reply]
Hey @nastja4 thanks for helping. The DNS check is green, which seems good news. But there's still this error message that is not very encouraging, but I'll give it some time to update.


>  Certificate Request Error: Certificate provisioning will retry automatically in a short period, please be patient.



<img width="1643" height="1103" alt="image" src="https://github.com/user-attachments/assets/a4bc45f0-1c76-4436-9fd9-8ce84ea5adcb" />

**Reply by gmagno** [reply]
Hello again,

Still no luck. Any other ideas? :pray: 

<img width="1100" height="1073" alt="image" src="https://github.com/user-attachments/assets/b5c84c66-c8d3-4f43-a447-5677cad13562" />

**Reply by gmagno** [reply]
If I refresh the github settings page, for a brief moment it shows: `Authorization created`. But then it shows the error message I shared in my previous post.

<img width="1181" height="1077" alt="image" src="https://github.com/user-attachments/assets/aa6baead-0015-4fb3-8bd9-d7bf4f9ba318" />

**Reply by nastja4** [reply]
Hi, could you please check the information above once more carefully, whether everything is the same as I had it?
Can you check the box 'Enforce HTTPS'?
Maybe use this advice from Dylan. 
Perhaps there are answers on https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site.

<img width="659" height="97" alt="image" src="https://github.com/user-attachments/assets/d7d504a0-cf0c-4715-b07e-b37c0a6456cb" />

**Reply by gmagno** [reply]
Hey, once again, thanks for trying to help, really appreciate it. I've read the information in this post several times.
I can't check the "Enforce HTTPS" box (see the image below).

<img width="1715" height="252" alt="image" src="https://github.com/user-attachments/assets/aaab07a0-b1b7-4783-b806-1bfd20397372" />

-----------

I've been also deploying the app every time I make changes:

<img width="1894" height="831" alt="image" src="https://github.com/user-attachments/assets/c6899896-a80e-4a41-b15d-0a79decbbd03" />

-----------

Thanks for the reference, I will do a bit more reading.

**Reply by nastja4** [reply]
Maybe ask GitHub Support?
